File: solutions/228.summary-ranges/summary-ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> res = new ArrayList<String>();
        if(nums.length < 1) return res;
        int start = 0, end = 0;
      
        for(int i = 1; i < nums.length; i++){
            if(nums[i] - nums[i-1] == 1){
                end ++;
            }else{
                res.add(start == end ? ""+nums[start] : nums[start] + "->" + nums[end]);
                start =  i;
                end = i;
            }
        }
        res.add(start == end ? ""+nums[start] : nums[start] + "->" + nums[end]);
        return res;
            
    }
        
}

File: solutions/228.summary-ranges/summary-ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
    if(nums.length === 0) return [];
    const ranges = [];
    let start = 0;
    for(let i = 0; i < nums.length; i++){
        if(i > 0 && nums[i] - nums[i - 1] !== 1){
            ranges.push(generateRange(nums, start, i - 1));
            start = i;
        }
    }
    ranges.push(generateRange(nums, start, nums.length - 1));
    return ranges;
    
    
    function generateRange(nums, start, end){
        if(start === end){
            return '' + nums[start];
        }else{
            return `${nums[start]}->${nums[end]}`;
        }
    }
    
};